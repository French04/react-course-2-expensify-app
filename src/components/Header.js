import React from 'react';
import {NavLink} from 'react-router-dom';
import {startLogout} from '../actions/auth';
import { connect } from 'react-redux';

export const Header = ({startLogout}) => (
    <header>
        <h1>Expensify</h1>
        <p><NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink></p>
        <p><NavLink to="create" activeClassName="is-active">Create Expense</NavLink></p>
        <button onClick={startLogout}>Logout</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);