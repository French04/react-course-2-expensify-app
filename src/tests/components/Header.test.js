import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../components/Header';

test('renderer Header correctly test', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
});
