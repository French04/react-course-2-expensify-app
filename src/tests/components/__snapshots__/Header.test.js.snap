// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer Header correctly test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    startLogout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Expensify
</h1>,
        <p>
          <NavLink
                    activeClassName="is-active"
                    ariaCurrent="true"
                    to="/dashboard"
          >
                    Dashboard
          </NavLink>
</p>,
        <p>
          <NavLink
                    activeClassName="is-active"
                    ariaCurrent="true"
                    to="create"
          >
                    Create Expense
          </NavLink>
</p>,
        <button
          onClick={[Function]}
>
          Logout
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expensify",
        },
        "ref": null,
        "rendered": "Expensify",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="/dashboard"
>
            Dashboard
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Dashboard",
            "to": "/dashboard",
          },
          "ref": null,
          "rendered": "Dashboard",
          "type": [Function],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="is-active"
            ariaCurrent="true"
            to="create"
>
            Create Expense
</NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "is-active",
            "ariaCurrent": "true",
            "children": "Create Expense",
            "to": "create",
          },
          "ref": null,
          "rendered": "Create Expense",
          "type": [Function],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Expensify
</h1>,
          <p>
            <NavLink
                        activeClassName="is-active"
                        ariaCurrent="true"
                        to="/dashboard"
            >
                        Dashboard
            </NavLink>
</p>,
          <p>
            <NavLink
                        activeClassName="is-active"
                        ariaCurrent="true"
                        to="create"
            >
                        Create Expense
            </NavLink>
</p>,
          <button
            onClick={[Function]}
>
            Logout
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expensify",
          },
          "ref": null,
          "rendered": "Expensify",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="is-active"
              ariaCurrent="true"
              to="/dashboard"
>
              Dashboard
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "is-active",
              "ariaCurrent": "true",
              "children": "Dashboard",
              "to": "/dashboard",
            },
            "ref": null,
            "rendered": "Dashboard",
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="is-active"
              ariaCurrent="true"
              to="create"
>
              Create Expense
</NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "is-active",
              "ariaCurrent": "true",
              "children": "Create Expense",
              "to": "create",
            },
            "ref": null,
            "rendered": "Create Expense",
            "type": [Function],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
